runner/client/src/mill/runner/client/MillProcessLauncher.java

```diff
@@ -106,15 +106,17 @@ static boolean isWin() {
   }
 
   static String javaExe() {
-    final String javaHome = System.getProperty("java.home");
-    if (javaHome != null && !javaHome.isEmpty()) {
-      final File exePath = new File(
-          javaHome + File.separator + "bin" + File.separator + "java" + (isWin() ? ".exe" : ""));
-      if (exePath.exists()) {
-        return exePath.getAbsolutePath();
-      }
+    String javaHome = System.getProperty("java.home");
+    String jvmId = "temurin:17";
```

Comment:
FYI, in Scala CLI (and also maybe in the coursier CLI), there's a [check of the version of the local JVM](https://github.com/VirtusLab/scala-cli/blob/51bebb087f9adaf1f1f4760374feb1a212b63bc9/modules/core/src/main/scala/scala/build/internals/OsLibc.scala#L89) before downloading one. If the version of the local one is high enough, it just uses it rather than downloading a new one.